src/weatherSource/WeatherSource.js

```diff
@@ -69,15 +64,17 @@ class OpenWeatherMapApi {
     // metoda prywatna zwracająca potrzebne dane o 
     // obecnej pogodzie z otrzymanego obiektu API
     _constructCurrentWeatherObject(data) {
-        var dataPrecipitationObject = this._getPrecipitationFromDataModel(data)
-
+        let dataPrecipitationObject = this._getPrecipitationFromDataModel(data)
+        let dataLocation = new Location(data.coord.lat, data.coord.lon, data.name, data.sys.country ? data.sys.country : "None")
```

Comment:
Location nie wymaga nazwy państwa (ani miasta, skoro przy tym jesteśmy). Jeśli nie masz żadnego państwa do przekazania do konstruktora, możesz go pominąć albo podać undefined. A kiedy piszę "możesz" to mam na myśli "powinieneś", bo inaczej None się będzie pojawiało jako nazwa państwa, a miasta jednak sa w państwach nawet, jeśli API nie zawsze takowe podaje.